src/quote/flashmint/hyeth/swap-data.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { AddressZero, EthAddress } from 'constants/addresses'
+import { noopSwapData } from 'constants/swapdata'
+import { ETH, USDC, WETH } from 'constants/tokens'
+import { QuoteToken } from 'quote/interfaces'
+import { Exchange, SwapData } from 'utils'
+
+export function getComponentsSwapData(isMinting: boolean): SwapData[] {
+  const path = isMinting
+    ? [WETH.address!, USDC.address!]
+    : [USDC.address!, WETH.address!]
+  return [
+    noopSwapData,
+    noopSwapData,
+    noopSwapData,
+    noopSwapData,
+    noopSwapData,
+    {
+      exchange: Exchange.UniV3,
+      fees: [500],
+      path,
+      pool: AddressZero,
+    },
+  ]
+}
+
+export function getEthToInputOutputTokenSwapData(
+  inputOutputToken: QuoteToken
+): SwapData | null {
+  if (inputOutputToken.symbol === WETH.symbol) {
+    return {
+      path: [EthAddress, inputOutputToken.address],
+      fees: [],
+      pool: AddressZero,
+      exchange: Exchange.None,
+    }
+  }
+  if (inputOutputToken.symbol === USDC.symbol) {
```

Comment:
Now that I see this function again. If I make another check for `ETH` to return null, I can in this case probably just always return with path `[WETH.address!, inputOutputToken.address]` no matter the ERC-20 input/output token, correct? @ckoopmann 

This would make it more future proof.